### PR TITLE
Display ip_addr_type as read-only when editing

### PR DESCRIPTION
**Pull request by kofrezo**
We still don't support changing the ip_addr_type of existing objects but
not being able to lookup the current type at all is not helpful so we
make it read-only in the edit form.

### CODE REVIEW

File: serveradmin/serverdb/admin.py [hunk 1]
```diff
@@ -31,12 +31,15 @@ class ServertypeAdmin(admin.ModelAdmin):
...
-        # Because of the great complexity when changing servertypes of existing
-        # objects and the little use-cases we have we just deny it for now.
+    def get_readonly_fields(self, request, obj=None):
         if obj:
+            # Because of the complexity when changing servertypes of existing
+            # objects and the little use-cases we have right now we don't
+            # support it.
             return ['ip_addr_type']
 
+        return super().get_readonly_fields(request, obj)
```
Comment (TheFRedFox) [inline serveradmin/serverdb/admin.py:41]: You first want to get the value of the super method and then add your own on it.
``` python
def get_readonly_fields(self, request, obj=None):
    fields = super().get_readonly_fields(request, obj)
    #fields = super().get_readonly_fields(request, obj) or []
    if obj:
        fields.append('ip_addr_type')

    return fields
```

**Review by TheFRedFox — CHANGES_REQUESTED** [review]
One detail which should be changed first.

File: serveradmin/serverdb/admin.py (hunk 1 above)
Comment (kofrezo) [reply serveradmin/serverdb/admin.py:41]: Ah yes that makes sense 👍

File: serveradmin/serverdb/admin.py [hunk 2]
```diff
@@ -31,11 +31,16 @@ class ServertypeAdmin(admin.ModelAdmin):
...
-        # objects and the little use-cases we have we just deny it for now.
+    def get_readonly_fields(self, request, obj=None):
+        fields = super().get_readonly_fields(request, obj)
+
         if obj:
-            return ['ip_addr_type']
+            # Because of the complexity when changing servertypes of existing
+            # objects and the little use-cases we have right now we don't
+            # support it.
+            fields = fields + tuple(['ip_addr_type'])
```
Comment (TheFRedFox) [inline serveradmin/serverdb/admin.py:41]:
```suggestion
            fields += ('ip_addr_type',)
```
or 
``` suggestion
            fields += tuple('ip_addr_type')
```
is possible.

**Review by TheFRedFox — APPROVED** [review]
One little suggestion, early approve.

**Pull request merged**